internal/cf/instance.go

```diff
@@ -49,6 +49,32 @@ func (io *instanceFilterOwner) getListOptions() *cfclient.ServiceInstanceListOpt
 // The function add the parameter values to the orphan cf instance, so that can be adopted.
 func (c *spaceClient) GetInstance(ctx context.Context, instanceOpts map[string]string) (*facade.Instance, error) {
 
+	if c.resourceCache.IsResourceCacheEnabled() {
+		// Ensure resourcesCache is initialized
+		if c.resourceCache == nil {
```

Comment:
You are checking resourceCache.IsResourceCacheEnabled and if true, check for resourceCache for nil.
The check for nil looks useless in the first view. Your coding still works because you implemented IsResourceCacheEnabled to deal with resourceCache being nil.
However, it is still confusing and unexpected.

Maybe you could just keep the pointer resourceCache nil, if resource caching is disabled.
All coding accessing the cache would need to check for nil then.
Then you could also remove the isResourceCacheEnabled variable inside the cache itself.

Reply:
Hi, Removed the condition